### ISSUE TITLE
[KED-2599] `kedro pipeline package` includes extraneous parameters files if another modular pipeline whose name is a superstring exists

### ISSUE THREAD

**Issue by deepyaman** [quantumblacklabs/kedro#759]
## Description
If you have two pipelines, `pipeline` and `pipeline_super`, and you package `pipeline`, `pipeline_super`'s parameters will get included because `pipeline_super` is a superstring of `pipeline`.

## Context
How has this bug affected you? What were you trying to accomplish?

I was trying to better understand/leverage the packaging mechanism, saw https://github.com/quantumblacklabs/kedro/blob/a6c47fa0722bb8c5e8b5b4ad236373cc5a99a850/kedro/framework/cli/pipeline.py#L468, and figured it would break for us ("retail" and "retail_banking" pipelines).

## Steps to Reproduce
```bash
conda create -n break-package python=3.8 -y
conda activate break-package
pip install 'kedro==0.17.3'
kedro new  # Press enter 3 times
cd new-kedro-project/
kedro install
kedro pipeline create retail
kedro pipeline create retail_banking
kedro pipeline package retail
unzip src/dist/retail-0.1-py3-none-any.whl 
```

## Expected Result
```bash
Archive:  src/dist/retail-0.1-py3-none-any.whl
  inflating: retail/README.md        
  inflating: retail/__init__.py      
  inflating: retail/nodes.py         
  inflating: retail/pipeline.py      
  inflating: retail/config/parameters/retail.yml  
  inflating: tests/__init__.py       
  inflating: tests/test_pipeline.py  
  inflating: retail-0.1.dist-info/METADATA  
  inflating: retail-0.1.dist-info/WHEEL  
  inflating: retail-0.1.dist-info/top_level.txt  
  inflating: retail-0.1.dist-info/RECORD  
```

## Actual Result
```bash
Archive:  src/dist/retail-0.1-py3-none-any.whl
  inflating: retail/README.md        
  inflating: retail/__init__.py      
  inflating: retail/nodes.py         
  inflating: retail/pipeline.py      
  inflating: retail/config/parameters/retail.yml  
  inflating: retail/config/parameters/retail_banking.yml  # Shouldn't be included
  inflating: tests/__init__.py       
  inflating: tests/test_pipeline.py  
  inflating: retail-0.1.dist-info/METADATA  
  inflating: retail-0.1.dist-info/WHEEL  
  inflating: retail-0.1.dist-info/top_level.txt  
  inflating: retail-0.1.dist-info/RECORD  
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): `0.17.3`
* Python version used (`python -V`): `3.8`
* Operating system and version:
```bash
(break-package) WASH-178551-C02X31K9JHD4:new-kedro-project deepyaman$ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 10.14.6 (18G9028)
      Kernel Version: Darwin 18.7.0
      ...
```

**Comment by yetudada** [comment]
Thank you so much for reporting this! I've logged it to our backlog to address.

**Comment by MerelTheisenQB** [comment]
Hi @deepyaman, a fix for this has now been merged and will be released in Kedro 0.17.4.
